OpenRA.Test/PathfinderTests.cs

```diff
@@ -0,0 +1,273 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Drawing;
+using System.Linq;
+using Moq;
+using NUnit.Framework;
+using OpenRA;
+using OpenRA.Mods.Common.Pathfinder;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Test;
+
+namespace PathfinderTests
+{
+	[TestFixture]
+	public class PathfinderTests
+	{
+		const int Width = 128;
+		const int Height = 128;
+		IWorld world;
+		IMap map;
+		IActor actor;
+
+		[SetUp]
+		public void Setup()
+		{
+			map = new FakeMap(Width, Height);
+			var worldactor = new FakeActor();
+			world = new FakeWorld(map, worldactor);
+			actor = new FakeActor(world);
+		}
+
+		IMap BuildFakeMap(int mapWidth, int mapHeight)
+		{
+			var map = new Mock<IMap>();
+			map.SetupGet(m => m.TileShape).Returns(TileShape.Rectangle);
+			map.Setup(m => m.MapSize).Returns(new int2(mapWidth, mapHeight));
+			map.Setup(m => m.Contains(It.Is<CPos>(pos => pos.X >= 0 && pos.X < mapWidth && pos.Y >= 0 && pos.Y < mapHeight))).Returns(true);
+
+			return map.Object;
+		}
+
+		IWorld BuildFakeWorld(IMap map)
+		{
+			var world = new Mock<IWorld>();
+			world.SetupGet(m => m.Map).Returns(map);
+			world.SetupGet(m => m.WorldActor).Returns(new Mock<IActor>().Object);
+			return world.Object;
+		}
+
+		static bool IsValidPos(CPos pos, int mapWidth, int mapHeight)
+		{
+			return pos.X >= 0 && pos.X < mapWidth && pos.Y >= 0 && pos.Y < mapHeight;
+		}
+
+		[Test]
+		[Ignore]
+		public void FindPathOnRoughTerrainTest()
+		{
+			// Arrange
+
+			// Create the MobileInfo Mock. Playing with this can help to
+			// check the different paths and points a unit can walk into
+			var mi = new FakeMobileInfo(pos => !(!IsValidPos(pos, Width, Height) ||
+					(pos.X == 50 && pos.Y < 100) ||
+					(pos.X == 100 && pos.Y > 50)));
```

Comment:
I guess that this this a good start, but it's nowhere near realistic.  We should start thinking about ways to define more realistic terrain data so that future prs can introduce more subtle tests.